pyomo/solvers/plugins/solvers/GUROBI.py

```diff
@@ -588,3 +594,233 @@ def _postsolve(self):
         TempfileManager.pop(remove=not self._keepfiles)
 
         return results
+
+
+@SolverFactory.register(
+    '_gurobi_file', doc='LP/MPS file-based direct interface to the GUROBI LP/MIP solver'
+)
+class GUROBIFILE(GUROBISHELL):
+    """Direct LP/MPS file-based interface to the GUROBI LP/MIP solver"""
+
+    def available(self, exception_flag=False):
+        if not gurobipy_available:  # this triggers the deferred import
+            if exception_flag:
+                raise ApplicationError("gurobipy module not importable")
+            return False
+        if getattr(self, '_available', None) is None:
+            self._check_license()
+        ans = self._available[0]
+        if exception_flag and not ans:
+            raise ApplicationError(msg % self.name)
+        return ans
+
+    def license_is_valid(self):
+        return self.available(False) and self._available[1]
+
+    def _check_license(self):
+        licensed = False
+        try:
+            # Gurobipy writes out license file information when creating
+            # the environment
+            with capture_output(capture_fd=True):
+                m = gurobipy.Model()
+            licensed = True
+        except gurobipy.GurobiError:
+            licensed = False
+
+        self._available = (True, licensed)
+
+    def _get_version(self):
+        return (
+            gurobipy.GRB.VERSION_MAJOR,
+            gurobipy.GRB.VERSION_MINOR,
+            gurobipy.GRB.VERSION_TECHNICAL,
+        )
+
+    def _default_executable(self):
+        # Bogus, but not None (because the test infrastructure disables
+        # solvers where the executable() is None)
+        return ""
+
+    def create_command_line(self, executable, problem_files):
+        #
+        # Define log file
+        #
+        if self._log_file is None:
+            self._log_file = TempfileManager.create_tempfile(suffix='.gurobi.log')
+
+        #
+        # Define command line
+        #
+        return Bunch(cmd=[], script="", log_file=self._log_file, env=None)
+
+    def _apply_solver(self):
+        #
+        # Execute the command
+        #
+        if is_debug_set(logger):
+            logger.debug("Running %s", self._command.cmd)
+
+        problem_filename = self._problem_files[0]
+        warmstart_filename = self._warm_start_file_name
+
+        # translate the options into a normal python dictionary, from a
+        # pyutilib SectionWrapper - because the gurobi_run function was
```

Comment:
I am pretty sure I asked this same question!

Reply:
Not quite.  `options` is a `Bunch` (which used to be a PyUtilib class).  The translation is necessary because `repr(Bunch)` doesn't produce meaningful python code.  Comment updated (and code simplified).